File: server/src/Wordki.Core/Mappers/WordMapper.cs
using System.Collections.Generic;
using Wordki.Core.Domain;
using Wordki.Core.Dtos;
using Wordki.Utils.Domain;

namespace Wordki.Core.Mappers
{
    public class WordMapper : IMapper<WordDto, Word>
    {
        private readonly IMapper<RepeatDto, Repeat> repeatMapper;

        public WordMapper(IMapper<RepeatDto, Repeat> repeatMapper)
        {
            this.repeatMapper = repeatMapper;
        }

        public Word Map(WordDto dto)
        {
            var drawer = Drawer.Restore(dto.Drawer);
            var repeats = new List<Repeat>();
            foreach (var repeatDto in dto.Repeats)
            {
                repeats.Add(repeatMapper.Map(repeatDto));
            }
            var word = Word.Restore(
                dto.WordId,
                dto.GroupId,
                dto.WordLanguage1,
                dto.WordLanguage2,
                dto.Example1,
                dto.Example2,
                dto.Comment,
                drawer,
                dto.IsVisible,
                dto.WordCreationDate,
                dto.NextRepeat,
                repeats);

            return word;
        }
    }
}

File: client/src/app/group-details/services/word-mapper/word-mapper.ts
import { Injectable } from '@angular/core';
import { WordDto } from '../../models/word.dto';
import { Word } from '../../models/word.model';

@Injectable()
export class WordMapper {
    map(dto: WordDto): Word {
        const word = new Word(dto.id,
            dto.word1.value,
            dto.word2.value,
            dto.word1.example,
            dto.word2.example,
            dto.word1.drawer,
            true,
            new Date());
        return word;
    }
}
